### DISCUSSION TITLE
Module-aware template variables

### DISCUSSION THREAD

**Discussion by op-ct** [puppetlabs-toy-chest/pct#132] [Ideas]
When I run commands like `pct new puppet-class` inside a module directory, it would be great if there was some way for the PCT template detect and use the actual module name instead of a static string (that I then have to go find and change everywhere).

Right now, if I create a new module and then a new class inside that module:

```sh
pct new puppet-module-base --name foo
cd foo
pct new puppet-class --name bar
```
...then the module name in the generated class manifest is the static string `module_name::` (straight from the [baker-round config][0]).

```console
[op-ct@host foo]$ grep -r :: manifests/ spec/classes/
manifests/bar.pp:class module_name::bar {
spec/classes/bar_spec.rb:describe 'module_name::bar' do
```

[0]: https://github.com/puppetlabs/baker-round/blob/main/puppet-class/pct-config.yml#L10

**Comment by op-ct** [comment]
I see that #32 is asking for the same thing.  Missed that the first time through.

**Comment by jpogran** [comment]
Thanks for the feedback @op-ct! This is something we are trying to address in a couple different ways. We recently added https://github.com/puppetlabs/pdkgo/pull/115, which supports a 'workspace file' inside a target project. It has a set of additional data you would like to have used inside templates. In this example, the module name would be there to be used at deploy time.

Why add another config file? We're trying to *not* add context/domain specific logic like this to the PCT tool itself. This means that we don't need to add a set of code for every new piece of data or a type of content. This means less change in the PCT tool which means less waiting for us to release a new PCT in order for you to build new content. We're trying to balance this goal with making scenarios like this ( a correct module name paired with a class name) intuitive and easy to accomplish.

**Comment by sanfrancrisko** [comment]
For additional context @op-ct , here's an example of how this is implemented: https://github.com/puppetlabs/baker-round/blob/d486db7e7223fd381a08ae984924a95a249480b4/puppetlabs/puppet-module-base/0.1.0/content/pct.yml.tmpl#L1-L2

The `puppet-module-base` project template will create a `pct.yml` that can be consumed by the `puppet-class` item template after. This pattern can be applied for similar use cases.
